applications/AISub.md

```diff
@@ -0,0 +1,256 @@
+- **FTE:**  4
+- **Costs:** 15,000 USD
+
+| Number | Deliverable                                                       | Specification                                                                                                                                                                                                                                                                                                                            |
+|-------:|-------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+|    0a. | License                                                           | GPL-3.0                                                                                                                                                                                                                                                                                                                                  |
+|    0b. | Documentation                                                     | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can (for example) spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works.                                                                                            |
+|    0c. | Testing Guide                                                     | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests.                                                                                                                                                                                        |
+|    0d. | Docker                                                            | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone.                                                                                                                                                                                                                            |
+|     1. | pallet-member                                                     | `create_membership_plan()`: Create new membership plans (admin only)<br>`purchase_membership_plan()`: Allow users to purchase a membership plan<br>`join_membership_plan_as_supplier()`: Suppliers join a membership plan to offer their models<br>`join_membership_plan_as_machine_supplier()`: Machine suppliers join a membership plan to perform calculations for models and receive pool rewards |
```

Comment:
One of the main features of memberships are that you are not allowed to share them without restrictions. How would you enforce this feature in a blockchain environment?

Reply:
In a blockchain environment, the actual membership duration is represented by the number of blocks. Once a user joins a membership plan, the membership start and end times are stored in the data. The end of the membership is determined based on these block numbers.
To enforce this feature, we use on_finalize to traverse the membership table and check whether the membership has reached the required number of blocks. If it has, the membership expires. Alternatively, each time a membership action is performed, the system checks if the stored membership expiration block number is less than the current block height. If so, membership operations will be restricted.